source/acquaman/SXRMB/SXRMBXRFScanController.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2010-2012 Mark Boots, David Chevrier, and Darren Hunter.
+Copyright 2013-2014 David Chevrier and Darren Hunter.
+
+This file is part of the Acquaman Data Acquisition and Management framework ("Acquaman").
+
+Acquaman is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Acquaman is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Acquaman.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#include "SXRMBXRFScanController.h"
+
+#include <QDir>
+
+
+#include "analysis/AMDeadTimeAB.h"
+#include "analysis/AM1DSummingAB.h"
+#include "dataman/AMUser.h"
+#include "dataman/datasource/AMRawDataSource.h"
+
+#include "application/SXRMB/SXRMB.h"
+#include "beamline/SXRMB/SXRMBBeamline.h"
+
+
+SXRMBXRFScanController::~SXRMBXRFScanController(){}
+
+SXRMBXRFScanController::SXRMBXRFScanController(SXRMBXRFScanConfiguration *scanConfig, QObject *parent)
+	: AMScanController(scanConfig, parent)
+{
+	detector_ = qobject_cast<AMXRFDetector *>(SXRMBBeamline::sxrmb()->exposedDetectorByInfo(scanConfig->detectorConfigurations().at(0)));
+
+	scan_ = new AMXRFScan;
+	scan_->setScanConfiguration(scanConfig);
+	scan_->setName(QString("XRF Scan - %1 el").arg(detector_->elements()));
+
+	scan_->setFilePath(AMUserSettings::defaultRelativePathForScan(QDateTime::currentDateTime()) + ".dat");
+	scan_->setFileFormat("vespers2011XRF");
+	scan_->setRunId(AMUser::user()->currentRunId());
+
+	int elements = detector_->elements();
+
+	for (int i = 0; i < elements; i++){
+
+		scan_->rawData()->addMeasurement(AMMeasurementInfo(QString("raw%1").arg(i+1), QString("Element %1").arg(i+1), "eV", detector_->axes()));
+		scan_->addRawDataSource(new AMRawDataSource(scan_->rawData(), i), true, false);
+	}
+
+	if (detector_->inputCountSources().size() > 0) {
+		for (int i = 0; i < elements; i++){
+
+			scan_->rawData()->addMeasurement(AMMeasurementInfo(QString("icr%1").arg(i+1), QString("Input count rate %1").arg(i+1), "%", QList<AMAxisInfo>()));
+			scan_->addRawDataSource(new AMRawDataSource(scan_->rawData(), i+elements), false, true);
+		}
+	}
+
+	if (detector_->outputCountSources().size() > 0) {
+		for (int i = 0; i < elements; i++){
+
+			scan_->rawData()->addMeasurement(AMMeasurementInfo(QString("ocr%1").arg(i+1), QString("Output count rate %1").arg(i+1), "%", QList<AMAxisInfo>()));
+			scan_->addRawDataSource(new AMRawDataSource(scan_->rawData(), i+2*elements), false, true);
+		}
+	}
+
+	for (int i = 0; i < elements; i++){
+
```

Comment:
The dead time analysis block shouldn't be used either if the ICR/OCR sources don't exist.